protocol/contracts/fakes/templegold/SpiceAuctionMock.sol

```diff
@@ -0,0 +1,125 @@
+    /// @notice Maximum auction duration
+    uint32 public constant MAXIMUM_AUCTION_DURATION = 30 days;
+    uint32 private immutable _arbitrumOneLzEid;
+    uint32 private immutable _mintChainId;
+    uint32 public lzReceiveExecutorGas;
+
+    /// @notice If true, let first claim for an auction notify total TGLD redeemed
+    bool public claimShouldNotifyTotalRedeemed;
+
+    /// @notice Name of this Spice Bazaar auction
+    string public name;
+
+    /// @notice Last time auction was started. For zero auctions, it is the contract deploy timestamp
+    uint256 private immutable _deployTimestamp;
+
+    event RedeemedTempleGoldBurned(uint256 amount);
+    error ExpectedNonZero();
+    error WithdrawFailed(uint256 amount);
+
+    constructor(
+        address _templeGold,
+        uint32 _arbEid,
+        uint32 mintChainId_,
+        string memory _name
+    ) {
+        templeGold = _templeGold;
+        _arbitrumOneLzEid = _arbEid;
+        _mintChainId = mintChainId_;
+        name = _name;
+        _deployTimestamp = block.timestamp;
+        lzReceiveExecutorGas = 85_412;
+    }
+
+    /// @notice withdraw ETH used for layer zero sends
+    function withdrawEth(address payable _to, uint256 _amount) external {
+        (bool success,) = _to.call{ value: _amount }("");
+        if (!success) { revert WithdrawFailed(_amount); }
+    }
+
+    /**
+     * @notice Burn redeemd TGLD and notify circulating supply
+     * @param amount Amount
+     * @param useContractEth If to use contract eth for layerzero send
+     */
+    function burnAndNotify(uint256 amount, bool useContractEth) external payable {
+        if (amount == 0) { revert ExpectedNonZero(); }
+        
+        emit RedeemedTempleGoldBurned(amount);
+        _burnAndNotify(amount, useContractEth);
+    }
+
+    function _burnAndNotify(uint256 amount, bool useContractEth) private {
+        // burn directly and call TempleGold to update circulating supply
+        if (block.chainid == _mintChainId) {
+            ITempleGold(templeGold).burn(amount);
+            return;
+        }
+        bytes memory options = OptionsBuilder.newOptions().addExecutorLzReceiveOption(lzReceiveExecutorGas, 0);
+        SendParam memory sendParam = SendParam(
+            _arbitrumOneLzEid, //<ARB_EID>,
+            bytes32(uint256(uint160(address(0)))), // bytes32(address(0)) to burn
+            amount,
+            0,
+            options,
+            bytes(""), // compose message
+            ""
+        );
+        MessagingFee memory fee = ITempleGold(templeGold).quoteSend(sendParam, false);
+        if (useContractEth && address(this).balance < fee.nativeFee) {
+            revert CommonEventsAndErrors.InsufficientBalance(address(0), fee.nativeFee, address(this).balance);
+        } else if (!useContractEth && msg.value < fee.nativeFee) {
+            revert CommonEventsAndErrors.InsufficientBalance(address(0), fee.nativeFee, msg.value); 
+        }
```

Comment:
I find this section a bit confusing, but perhaps I'm not understanding the purpose. 

When `msg.value > 0`, the `msg.value` is added to the contract balance before reaching this piece of code. So in the check:

```address(this).balance < fee.nativeFee```

you are actually checking for:

```address(this).balance{before} + msg.value < fee.nativeFee```

 Therefore, if `msg.value > 0` and `useContractEth=true`, you have a weird state in which you end up going through the upper branch if the if statement, but passing the requirement. You could argue that the caller should know what args to pass, and they I'd argue, "then simply let them pass the `msg.value` they need to make up for the total eth balance to be enough".

So, a simpler version of this logic is to only require `address(this).balance < fee.nativeFee`, and the sender could send `msg.value` to increase it if necessary. 

Then, regarding the logic below (who receives the leftover eth), you can pass an address as a receiver of the leftover eth. Then,  you can get rid of `useContractEth`. 

But perhaps I'm not fully understanding its purpose